File: Java-Core/src/main/java/prompto/javascript/JavaScriptThisExpression.java
package prompto.javascript;

import prompto.expression.ThisExpression;
import prompto.utils.CodeWriter;

public class JavaScriptThisExpression implements JavaScriptExpression {
	
	ThisExpression expression = new ThisExpression();
	
	@Override
	public void toDialect(CodeWriter writer) {
		expression.toDialect(writer);
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptThisExpression.js
var JavaScriptExpression = require("./JavaScriptExpression").JavaScriptExpression;
var ThisExpression = require("../expression/ThisExpression").ThisExpression;

function JavaScriptThisExpression() {
    JavaScriptExpression.call(this);
	this.expression = new ThisExpression();
	return this;
}

JavaScriptThisExpression.prototype = Object.create(JavaScriptExpression.prototype);
JavaScriptThisExpression.prototype.constructor = JavaScriptThisExpression;

JavaScriptThisExpression.prototype.interpret = function(context) {
    return this.expression.interpret(context);
};

JavaScriptThisExpression.prototype.toDialect = function(writer) {
    return this.expression.toDialect(writer);
};

JavaScriptThisExpression.prototype.toString = function() {
	return this.expression.toString();
};

exports.JavaScriptThisExpression = JavaScriptThisExpression;